### PR TITLE
chore: migrate to @angular/compiler-cli v5

### LINKED ISSUE #2122: Update to angular version 5.0.0
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x ] feature request
[ ] support request => Please do not submit support request here, instead see use [gitter](https://gitter.im/mgechev/angular2-seed) or [stackoverflow](https://stackoverflow.com/questions/tagged/angular2)
```
**What is the motivation / use case for changing the behavior?**

Angular latest release

### PR DESCRIPTION
**Pull request by mgechev**
Fix #2122

### CODE REVIEW

File: tools/tasks/seed/compile.ahead.prod.ts [hunk 1]
```diff
@@ -9,64 +10,64 @@ import { argv } from 'yargs';
...
+  args.push('-p', join(Config.TMP_DIR, 'tsconfig.json'));
+  Object.keys(namedArgs)
+    .filter(n => NgCompilerOptions.indexOf(n) >= 0)
+    .forEach((key: string) => {
+      if (key === 'lang') {
+        const lang: string = namedArgs[key] as string;
+        const i18nFilePath = `${Config.LOCALE_DEST}/messages.${lang}.xlf`;
+        const isExists = existsSync(i18nFilePath);
+        if (isExists) {
+          args.push('--i18n-file', i18nFilePath);
```
Comment (mikejohanson) [inline tools/tasks/seed/compile.ahead.prod.ts]: I think this needs to be: `args.push('--i18nFile', i18nFilePath);` as it has changed from --i18n-file.
https://angular.io/guide/i18n#merge-the-completed-translation-file-into-the-app

File: tools/tasks/seed/compile.ahead.prod.ts [hunk 2]
```diff
@@ -9,64 +10,64 @@ import { argv } from 'yargs';
...
+          args.push('--locale', lang);
+          args.push('--i18n-format', 'xlf');
```
Comment (mikejohanson) [inline tools/tasks/seed/compile.ahead.prod.ts]: This one should be `args.push('--i18nFormat', 'xlf');`

File: tools/tasks/seed/compile.ahead.prod.ts (hunk 1 above)
Comment (mgechev) [reply tools/tasks/seed/compile.ahead.prod.ts]: Thanks! You tested it with `--i18nFilePath` and it works properly?

Comment (mikejohanson) [reply tools/tasks/seed/compile.ahead.prod.ts]: I ran it with `npm run build.prod.aot -- --lang en-US` with the above change and it works properly. I'm not sure about `--i18nFilePath`  but `--i18nFile` , yes that works.

**Pull request merged**